app/gui/view/graph-editor/src/component/node/input/widget/single_choice.rs

```diff
@@ -69,17 +69,30 @@ pub struct Config {
     /// Entries that should be displayed by the widget, as proposed by language server. This
     /// list is not exhaustive. The widget implementation might present additional
     /// options or allow arbitrary user input.
-    pub entries:   Rc<Vec<Entry>>,
-    /// Optional widget configurations for arguments of a method call in entries expression.
-    /// The first element of the tuple is the index of an associated entry in `entries` vector,
-    /// and the vector is also ordered by it.
-    pub arguments: Vec<(usize, ImString, Configuration)>,
+    pub choices:   Rc<Vec<Choice>>,
+    /// Optional widget configurations for arguments for method or constructor applications within
+    /// choices' expressions. This list is always ordered by `choice_index`.
+    pub arguments: Vec<ChoiceArgConfig>,
+}
+
+/// An optional configuration override for an argument nested within a given dropdown choice. Only
+/// applies to the arguments of an application directly nested within the choice's expression. That
+/// includes method calls, standalone functions or constructors.
+#[derive(Debug, Clone, PartialEq)]
+pub struct ChoiceArgConfig {
+    /// The index of an associated choice in `choices` vector.
```

Comment:
I would link to the structure containing "choices" vector.
```suggestion
    /// The index of an associated choice in [`choices`](Config) vector.
```